api/src/main/java/edu/cornell/mannlib/vitro/webapp/dao/jena/RDFServiceGraph.java

```diff
@@ -94,37 +95,105 @@ private StringBuilder serialize(StringBuilder sb, Triple t) {
                .append(sparqlNodeUpdate(t.getObject(), "")).append(" .");
         return sb;
     }
+    
+    private String serialize(Triple t) {
+        StringBuffer sb = new StringBuffer();
+        sb.append(sparqlNodeUpdate(t.getSubject(), "")).append(" ") 
+               .append(sparqlNodeUpdate(t.getPredicate(), "")).append(" ") 
+               .append(sparqlNodeUpdate(t.getObject(), "")).append(" .");
+        return sb.toString();
+    }
 
     private synchronized void flush() {
         ChangeSet changeSet = rdfService.manufactureChangeSet();
         try {
             if(!removalsGraph.isEmpty()) {
-                String removals = serializeGraph(removalsGraph);
-                changeSet.addRemoval(RDFServiceUtils.toInputStream(removals),
-                        RDFService.ModelSerializationFormat.N3, graphURI);
-                removalsGraph.clear();
+                flushGraph(removalsGraph,graphURI,changeSet,rdfService,false);
             }
             if(!additionsGraph.isEmpty()) {
-                String additions = serializeGraph(additionsGraph);
-                changeSet.addAddition(RDFServiceUtils.toInputStream(additions),
-                        RDFService.ModelSerializationFormat.N3, graphURI);
-                additionsGraph.clear();
+                flushGraph(additionsGraph,graphURI,changeSet,rdfService,true);
             }
-            rdfService.changeSetUpdate(changeSet);
         } catch (RDFServiceException rdfse) {
             throw new RuntimeException(rdfse);
         }
     }
+    
+    
+    private static int maxStringSize = 10000000;
+    
+    private synchronized void flushGraph(Graph graph,String graphURI,ChangeSet changeSet,RDFService rdfService,boolean add) throws RDFServiceException {
```

Comment:
The `graphURI` is a class data member:
https://github.com/vivo-project/Vitro/blob/a369b339352fa569438e1680ba59e840134f7ba7/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dao/jena/RDFServiceGraph.java#L51

Maybe we can remove it from the method signature?